primitives/avn-common/src/eth/mod.rs

```diff
@@ -123,3 +159,107 @@ impl TryFrom<&[u8]> for BridgeContractMethod {
         }
     }
 }
+
+pub fn eip712_hash<T: SolStruct>(data: &T, domain: &Eip712Domain) -> H256 {
+    let hash: B256 = data.eip712_signing_hash(domain);
+    H256::from_slice(&hash.0)
+}
+
+fn parse_from_utf8<T: std::str::FromStr>(bytes: &[u8]) -> Result<T, ()> {
+    str::from_utf8(bytes).map_err(|_| ())?.parse::<T>().map_err(|_| ())
+}
+
+pub fn create_function_confirmation_hash(
```

Comment:
I think this function should take `data` and `domain` as a parameter and return the eip_712 hash. The logic of constructing the data should stay in the relevant pallets. And maybe we can call it `get_t1_tx_hash` ?